### PR TITLE
Sync feature - transfer subfeatures and global state to feature json

### PR DESCRIPTION
**Pull request by cmonfortep**
**Asana Task/Github Issue:** https://app.asana.com/0/0/1206190482102399/f

## Description
Decided to have a global config instead of a per platform override.

#### Reference
- [Config Reviewer Documentation](https://app.asana.com/0/1200890834746050/1204443212791216/f)
- [Config Maintainer Documentation](https://app.asana.com/0/1200890834746050/1200573250322769/f)
- [Feature Implementer Documentation](https://app.asana.com/0/1200890834746050/1201498956177210/f)

### CODE REVIEW

**Review by ayoy — COMMENTED** [review]
@cmonfortep could you please update iOS and macOS overrides to remove sync completely? That's because their current overrides are going to the top level definition with this PR. Thanks!

**Comment by cmonfortep** [comment]
oh I didn't know you already added them! you are so fast @ayoy

**Comment by ayoy** [comment]
It's actually @SlayterDev who is doing a wonderful job reviewing all the PRs within minutes 💪🏻💪🏻💪🏻

**Comment by ayoy** [comment]
@cmonfortep sorry to be a pain, but there's also the macos-override.json that needs to be updated. I'm AFK so can't do it myself at the moment. Thanks 😇

File: features/sync.json [hunk 1]
```diff
@@ -3,5 +3,20 @@
         "description": "Sync & Backup browser feature. Feature available on native clients. Allow users to sync bookmarks, logins, etc.",
         "sampleExcludeRecords": {}
     },
+    "state": "internal",
```
Comment (SlayterDev) [inline features/sync.json:6]: Hate to be a pain but we should probably default disable this on all platforms and just set it to internal where needed. We don't want any unintended side effects on other platforms even if they don't use the feature.

**Review by SlayterDev — CHANGES_REQUESTED** [review]

File: features/sync.json (hunk 1 above)
Comment (bwaresiak) [reply features/sync.json:6]: I wonder about that. Is 'internal' a state that is expected to be globally supported? If not, are there plans for that? Cause what I'd expect it to be able to use internal state exactly in situations like this - where we are developing something and we have it as an internal feature across products (for products that don't support given feature it should not matter - as they won't parse the flag in the first place).

Comment (cmonfortep) [reply features/sync.json]: In theory it should work like @bwaresiak said, but I also prefer to err on the side of being careful here and leave default state as `disabled`.  At least is consistent with the value when the state is not present.

**Review by SlayterDev — APPROVED** [review]

**Pull request merged**